tardis/opacities/opacity_solver.py

```diff
@@ -0,0 +1,55 @@
+from tardis.opacities.tau_sobolev import calculate_sobolev_line_opacity
+from tardis.opacities.opacity_state import (
+    OpacityState,
+)
+import numpy as np
+
+
+class OpacitySolver(object):
+
+    line_interaction_type: str = "scatter"
+    disable_line_scattering: bool = False
+
+    def __init__(
+        self, line_interaction_type="scatter", disable_line_scattering=False
+    ):
+        """Solver class for opacities
+
+        Parameters
+        ----------
+        line_interaction_type: str
+            "scatter", "downbranch", or "macroatom"
+        disable_line_scattering: bool
+        """
+
+        self.line_interaction_type = line_interaction_type
+        self.disable_line_scattering = disable_line_scattering
+
+    def solve(self, legacy_plasma) -> OpacityState:
+        """
+        Solves the opacity state
+
+        Parameters
+        ----------
+        plasma : tarids.plasma.BasePlasma
+            legacy base plasma
+
+        Returns
+        -------
+        OpacityState
+        """
+        tau_sobolev = calculate_sobolev_line_opacity(
+            legacy_plasma.atomic_data.lines,
+            legacy_plasma.level_number_density,
+            legacy_plasma.time_explosion,
+            legacy_plasma.stimulated_emission_factor,
+        )
+
+        if self.disable_line_scattering:
```

Comment:
It's quite odd to calculate this and then set it to 0 if it's disabled, and doesn't save on any computation. Can we just initialize this as 0s of the right shape instead?